test/scenarios/constantEffectInteractions/losingAndGainingKeywords.spec.ts

```diff
@@ -0,0 +1,56 @@
+describe('Losing and gaining keywords', function() {
+    integration(function(contextRef) {
+        describe('If a unit has a keyword given by a source, then loses that keyword due to an ability,', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['specforce-soldier'],
+                        groundArena: ['wampa', 'crafty-smuggler']
+                    },
+                    player2: {
+                        groundArena: ['vigilant-honor-guards'],
+                        hand: ['protector', 'repair']
+                    }
+                });
+            });
+
+            it('it should be able to gain the keyword again from a different source', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.specforceSoldier);
+                context.player1.clickCard(context.vigilantHonorGuards);
+
+                context.player2.clickCard(context.protector);
+                context.player2.clickCard(context.vigilantHonorGuards);
+
+                context.player1.clickCard(context.wampa);
+
+                // attack target chosen automatically due to sentinel
+                expect(context.player2).toBeActivePlayer();
+                expect(context.wampa).toBeInLocation('discard');
+                expect(context.vigilantHonorGuards.damage).toBe(4);
+            });
+
+            it('it should not regain the keyword from that source after the source\'s condition toggles off then on', function() {
```

Comment:
I am... actually unsure of what the SWU rules say should happen here? I may ask in the judge chat, it's not clear to me whether this situation counts as applying a new effect or not (obviously if it is a new effect then this test is wrong)

Reply:
Ah okay, looks like the test is right: https://discord.com/channels/1265343874105081856/1285715527548796968

Good job with this, would not have thought of this test at all